src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4988,22 +5053,28 @@ std::optional<ProjectionCandidate> MergeTreeData::getQueryProcessingStageWithAgg
 
         if (projection.type == ProjectionDescription::Type::Aggregate && analysis_result.need_aggregate && can_use_aggregate_projection)
         {
-            bool match = true;
             Block aggregates;
             // Let's first check if all aggregates are provided by current projection
             for (const auto & aggregate : select.getQueryAnalyzer()->aggregates())
             {
-                const auto * column = sample_block.findByName(aggregate.column_name);
-                if (column)
+                if (const auto * column = sample_block.findByName(aggregate.column_name))
+                {
                     aggregates.insert(*column);
-                else
+                    continue;
+                }
+
+                if (normalize_count_not_null && dynamic_cast<const AggregateFunctionCount *>(aggregate.function.get()))
                 {
-                    match = false;
-                    break;
+                    const auto * count_column = sample_block.findByName("count()");
```

Comment:
I see in the test, there are cases with `count` with arguments, like `count(if(d=4, d, 1))`. But here we try to find a column with name `count()`. So, I don't understand how it works - shouldn't the column name be `count(if(d=4, d, 1))` as well?

Reply:
The `sample_block` is from `minmax_count_projection` which can only have `count()` as the row count column. During projection analysis, all `count(not-null-expression)` will be normalized to `count()` to check if matched.